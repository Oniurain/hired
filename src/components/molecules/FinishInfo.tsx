import * as React from 'react'
import styled from 'styled-components'

import { addHours, formatTime } from '../../lib/dateTime'

import { color } from '../../styles/style-utils'
import { flip } from '../../styles/globals'

const Root = styled.div`
  animation: ${flip} 5s ease-in-out 2s infinite alternate both;
  position: relative;
  transform-style: preserve-3d;
`

const Start = styled.div`
  background-color: ${color.gray.g_50};
  backface-visibility: hidden;
  left: 0;
  position: absolute;
  top: 0;
  z-index: 20;
`

const Finish = styled.div`
  background-color: ${color.gray.g_50};
  backface-visibility: hidden;
  left: 0;
  position: absolute;
  top: 0;
  transform: rotateX(180deg);
  z-index: 10;
`

export interface FinishInfoProps {
  className?: string
  startTime: Date
  workHours: number
}

const FinishInfo = ({className, startTime, workHours}: FinishInfoProps) => {
  return (
    <Root className={className}>
      <Start>started at {formatTime(startTime)}</Start>
      <Finish>done at {formatTime(addHours(startTime, workHours))}</Finish>
    </Root>
  )
}

export default FinishInfo
