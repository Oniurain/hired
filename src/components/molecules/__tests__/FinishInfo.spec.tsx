import * as React from 'react'
import { mount, ReactWrapper } from 'enzyme'

import FinishInfo, { FinishInfoProps } from '../FinishInfo'

describe('FinishInfo', () => {
  let subject: ReactWrapper<FinishInfoProps, {}>

  beforeEach(() => {
    const startTime = new Date(2017, 1, 14, 10, 22)

    subject = mount(
      <FinishInfo
        startTime={startTime}
        workHours={8}
      />
    )

  })
  it('should render', () => {
    expect(subject).toMatchSnapshot()
  })
})
