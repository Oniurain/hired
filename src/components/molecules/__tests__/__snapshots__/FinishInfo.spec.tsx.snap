// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FinishInfo should render 1`] = `
<FinishInfo
  startTime={2017-02-14T09:22:00.000Z}
  workHours={8}
>
  <styled.div>
    <div
      className="icEpLt"
    >
      <styled.div>
        <div
          className="inEzxg"
        >
          started at 
          10:22
        </div>
      </styled.div>
      <styled.div>
        <div
          className="dmDyDa"
        >
          done at 
          18:22
        </div>
      </styled.div>
    </div>
  </styled.div>
</FinishInfo>
`;
